src/envoy/utils/utils.cc

```diff
@@ -47,16 +51,21 @@ bool getCertSAN(const Network::Connection* connection, bool peer,
         // empty result is not allowed.
         return false;
       }
+      // return the first san with the 'spiffe://' prefix
+      for (const auto& san : sans) {
+        if (hasSPIFFEPrefix(san)) {
+          *principal = san;
+          return true;
+        }
+      }
+      // return the first san if no sans have the spiffe:// prefix
       *principal = sans[0];
       return true;
     }
   }
   return false;
 }
 
-bool hasSPIFFEPrefix(const std::string& san) {
-  return absl::StartsWith(san, kSPIFFEPrefix);
```

Comment:
Is this the same as the change from line 39-42?

Reply:
Yeah it's the same. I think it will cause build error without moving it above.